Add widget class and some new dashboard widgets

Moppa5 opened the pull request:
#### What are the main changes in this PR?
Add Dashboard_Widgets class for wp-admin dashboard widgets class. Extend dashboard widgets class by adding PHP end of life version and low disk space notifications. Also, move PHP error count widget from login notifications to the dashboard widgets class together with the new site status widget. Includes translations. 

PHP EOL needs reworking when the common config file / internal implementation is done. 

##### Why are we doing this? Any context or related work?
See #517

#### Manual testing steps?
To test end of life, error count and low disk space widgets there are some requirements:
- Some PHP errors on log (artificial or not)
- Switch to PHP 7.2
- `const LOW_DISK_SPACE_USAGE = 0.9;` change the value to something like 0.1 for instance. See the line [here](https://github.com/Seravo/seravo-plugin/blob/9d37bbb5afa32bbb256733ed189e8e1635f54283/modules/dashboard-widgets.php#L24)

Note that disk space usage is stored as a transient for 15 minutes, thus you need to delete transients and purge cache if you want to re-calculate and usage results. 

#### Screenshots
PHP EOL and warnings + disk space widget. For site status widget see #508 
![Screenshot(3)](https://user-images.githubusercontent.com/16706187/114685315-6c807300-9d1a-11eb-9cf1-38a8c7bf3ae0.png)

JoosuaKoskinen commented on modules/dashboard-widgets.php:
```diff
@@ -0,0 +1,301 @@
+      if ( $return_code !== false ) {
+        $data_size = preg_split('/\s+/', $data_folder[0])[0];
+      }
+      $plan_details = API::get_site_data();
```
I'm bit worried about this and the `exec` above. A lot of slow things are done every time accessing dashboard. I saw a difference up to 50-100ms in load time and we want to be as fast as possible. I believe the load time might increase even more on sites with a lot of individual files for `du`to check.

`get_site_data` is called multiple times on dashboard load and it takes the most time on my site. Maybe the resulting `['plan']['disklimit']` should be cached into transient?

Moppa5 replied:
Good point! Caching sounds a good solution to this. It needs some reworking for now.

Moppa5 replied:
Now added transients for 1 hour both in dashboard widgets and site status page.

JoosuaKoskinen commented on modules/dashboard-widgets.php:
```diff
@@ -0,0 +1,301 @@
+
+      if ( $plan_disk_limit !== 0 && $data_size !== 0 ) {
+        $data_size_human = (int) Helpers::human_file_size($data_size);
+        $relative_disk_space_usage = $data_size_human / ($plan_disk_limit * 1000);
```
This relative disk space usage doesn't work. My site has 1.2GB in files with 1GB limit and the alert box didn't show up. It seems to assume `human_file_size` to always give the result in MB but it can give it any format. 

On my site, `human_file_size` returned `"1.2GB"` but as it's converted it into int it was `1` making this statement `$relative_disk_space_usage = 1 / (1 * 1000)` which is incorrect relative disk space usage.

See https://github.com/Seravo/seravo-plugin/blob/master/lib/helpers.php#L38

Moppa5 replied:
I was actually afraid that the helper method wouldn't work in the desired way. Need to re-work the calculations.

Moppa5 replied:
Re-worked the calculations and it should now apply the MBs and calculate the relative usage properly.

JoosuaKoskinen reviewed (changes requested):
LGTM otherwise!

Moppa5 commented:
@JoosuaKoskinen, I updated and squashed the branch. I added transients for 15 minutes to store the disk space usage and not calculate it all the time. I changed the logic how the relative disk space is calculated and it should work now as intended. Also, there are some minor changes and improvements + translations as well.

JoosuaKoskinen commented:
> What do you think is the new implemented one excessive?

I don't think so, no. The built-in PHP nag wasn't introduced until core 5.1.0 which there's still a lot of sites running on.

We either need to check if wp core version is >= 5.1.0 and hide the new widget or just hide the core built-in one completely. I think we should do the latter one as the instruction links in the built-in one doesn't apply in our environment (we wan't to link to https://help.seravo.com and PHP-changer widget).

Moppa5 commented:
@JoosuaKoskinen, Good points! I ended up hiding the WP PHP nag as mentioned that our is more Seravo specific. Transient case, calculations and the hiding are now done. Could you review whether we can merge the feature or not?

JoosuaKoskinen commented on modules/dashboard-widgets.php:
```diff
@@ -0,0 +1,328 @@
+            $bar_size = 1;
+          }
+
+          if ( $bar_size <= 10 ) {
+            $bar_css = 'auto';
+          } else {
+            $bar_css = strval($bar_size) . '%';
+          }
```
Even with this, the bar is sometimes too small and the content overflows. Maybe just remove this and add `min-width: $bar_size%; width: auto;` to the bar css so it's always at least wide enough to fit the content.

JoosuaKoskinen reviewed (changes requested):
Looks really good! Feel free to merge after resolving that one change request.
